pandas/core/indexes/interval.py

```diff
@@ -870,112 +859,140 @@ def get_value(self, series, key):
             None is specified as these are not yet implemented.
         """)}))
     @Appender(_index_shared_docs['get_indexer'])
-    def get_indexer(self, target, method=None, limit=None, tolerance=None):
+    def get_indexer(self,
+                    target: AnyArrayLike,
+                    method: Optional[str] = None,
+                    limit: Optional[int] = None,
+                    tolerance: Optional[Any] = None
+                    ) -> np.ndarray:
 
         self._check_method(method)
-        target = ensure_index(target)
-        target = self._maybe_cast_indexed(target)
 
-        if self.equals(target):
-            return np.arange(len(self), dtype='intp')
+        if self.is_overlapping:
+            msg = ('cannot handle overlapping indices; use '
+                   'IntervalIndex.get_indexer_non_unique')
+            raise InvalidIndexError(msg)
 
-        if self.is_non_overlapping_monotonic:
-            start, stop = self._find_non_overlapping_monotonic_bounds(target)
-
-            start_plus_one = start + 1
-            if not ((start_plus_one < stop).any()):
-                return np.where(start_plus_one == stop, start, -1)
-
-        if not self.is_unique:
-            raise ValueError("cannot handle non-unique indices")
+        try:
+            target = ensure_index(target)
+        except ValueError:
```

Comment:
hmm, why is this needed? ensure_index should just work here

Reply:
Looks like it `ensure_index` raises when provided a list of mixed closed intervals:
```python-traceback
In [6]: ivs = [pd.Interval(0, 1, closed='left'), 
   ...:        pd.Interval(1, 2, closed='right')]

In [7]: ensure_index(ivs)
---------------------------------------------------------------------------
ValueError: intervals must all be closed on the same side
```
I'm guessing we'd want `ensure_index` to have this return an `object` dtype `Index`?  Would we also want `Index(ivs)` to do the same?

Reply:
yep i think it guarantees to return an Index (which maybe object)

Reply:
Okay, found a fix; it's pretty small but will do it in a separate PR later tonight, as the scope of this PR is large enough already (unless you'd rather it be done here).

Reply:
can u do as a precursor ?

Reply:
this can now be removed (and below)